engine/security/src/main/scala/pl/touk/nussknacker/ui/security/api/LoggedUser.scala

```diff
@@ -11,19 +15,28 @@ sealed trait LoggedUser {
 }
 
 object LoggedUser {
-   def apply(id: String,
-             username: String,
-             categoryPermissions: Map[String, Set[Permission]] = Map.empty,
-             globalPermissions: List[GlobalPermission] = Nil,
-             isAdmin: Boolean = false): LoggedUser = {
-     if (isAdmin) {
-       AdminUser(id, username)
-     } else {
-       CommonUser(id, username, categoryPermissions, globalPermissions)
-     }
+  def apply(authenticatedUser: AuthenticatedUser, config: Config, processCategories: List[String]): LoggedUser = {
+    apply(authenticatedUser, AuthenticationConfiguration.getRules(config), processCategories)
+  }
+
+  def apply(authenticatedUser: AuthenticatedUser, rules: List[ConfigRule], processCategories: List[String]): LoggedUser = {
+    val rulesSet = RulesSet.getOnlyMatchingRules(authenticatedUser.roles, rules, processCategories)
+    apply(id = authenticatedUser.id, username = authenticatedUser.username, authenticatedUser.roles, rulesSet = rulesSet)
+  }
+
+  def apply(id: String,
+            username: String,
+            categoryPermissions: Map[String, Set[Permission]] = Map.empty,
+            globalPermissions: List[GlobalPermission] = Nil,
+            isAdmin: Boolean = false): LoggedUser = {
+    if (isAdmin) {
+      AdminUser(id, username)
+    } else {
+      CommonUser(id, username, categoryPermissions, globalPermissions)
+    }
   }
 
-  def apply(id: String, username: String, rulesSet: RulesSet): LoggedUser = {
+  def apply(id: String, username: String, roles: List[String], rulesSet: RulesSet): LoggedUser = {
```

Comment:
roles are not used here?

Reply:
Right. Garbage. I thought of passing roles along with permissions in LoggedUser but then dropped the idea. Removed.